examples/text/plot_hashing_vs_dict_vectorizer.py

```diff
@@ -3,109 +3,293 @@
+# as fast as using the `DictVectorizer` along with the simple function we
+# defined for mapping the tokens. The reason is that `CountVectorizer` produces
+# a sparse representation of the counts using `scipy.sparse.csr_matrix`.
+#
+# Now we make a similar experiment with the `HashingVectorizer`, which is
+# equivalent to combining the “hashing trick” implemented by the `FeatureHasher`
+# class and the text preprocessing and tokenization of the `CountVectorizer`.
+
+from sklearn.feature_extraction.text import HashingVectorizer
+
+t0 = time()
+vectorizer = HashingVectorizer(n_features=2**18)
+vectorizer.fit_transform(raw_data)
+duration = time() - t0
+dict_count_vectorizers["vectorizer"].append(vectorizer.__class__.__name__)
+dict_count_vectorizers["speed"].append(data_size_mb / duration)
+print(f"done in {duration:.3f}s at {data_size_mb / duration:.3f}MB/s")
+
+# %%
+# TfidfVectorizer
+# ---------------
+#
+# In a large text corpus, some words appear with higher frequency (e.g. “the”,
+# “a”, “is” in English) which do not carry meaningful information about the
+# actual contents of a document. If we were to feed the word count data directly
+# to a classifier those very common terms would shadow the frequencies of rarer
+# yet more interesting terms. In order to re-weight the count features into
+# floating point values suitable for usage by a classifier it is very common to
+# use the tf–idf transform as implemented by the `TfidfTransformer`. Tf stands
+# for "term-frequency" while "tf–idf" means term-frequency times inverse
+# document-frequency.
+#
+# We now benchmark the `TfidfVectorizer`, which is equivalent to combining the
+# tokenization and occurrence counting of the `CountVectorizer` along with the
+# normalizing and weighting from a `TfidfTransformer`.
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+t0 = time()
+vectorizer = TfidfVectorizer()
+vectorizer.fit_transform(raw_data)
+duration = time() - t0
+dict_count_vectorizers["vectorizer"].append(vectorizer.__class__.__name__)
+dict_count_vectorizers["speed"].append(data_size_mb / duration)
+print(f"done in {duration:.3f}s at {data_size_mb / duration:.3f}MB/s")
+print(f"Found {len(vectorizer.get_feature_names_out())} unique terms")
+
+fig, ax = plt.subplots(figsize=(12, 6))
+
+y_pos = np.arange(len(dict_count_vectorizers["vectorizer"]))
+ax.barh(y_pos, dict_count_vectorizers["speed"], align="center")
+ax.set_yticks(y_pos)
+ax.set_yticklabels(dict_count_vectorizers["vectorizer"])
+ax.invert_yaxis()
+_ = ax.set_xlabel("speed (MB/s)")
+
+# %%
+# Notice from the plot that `TfidfVectorizer` is slightly slower than
+# `CountVectorizer` because of the extra operation induced by the
+# `TfidfTransformer`. Also notice that, by setting the number of features
+# `n_features = 2**18`, the `HashingVectorizer` performs much faster than the
+# `TfidfVectorizer`. Keep in mind that this may cause hash collisions and no
```

Comment:
```suggestion
# `CountVectorizer`. Keep in mind that this may cause hash collisions and no
```

`HashingVectorizer` does not do any TF-IDF normalization, so it's better presented as an alternative to `CountVectorizer`.